src/main/java/no/ndla/taxonomy/repositories/NodeConnectionRepository.java

```diff
@@ -20,10 +20,16 @@
 import java.util.Set;
 
 public interface NodeConnectionRepository extends TaxonomyRepository<NodeConnection> {
-    @SuppressWarnings("SpringDataRepositoryMethodReturnTypeInspection")
-    @Query("SELECT nc.parent.id AS parentId, nc.child.id AS childId, nc.rank AS rank FROM NodeConnection nc"
-            + " JOIN nc.parent JOIN nc.child WHERE nc.parent.id IN :nodeId")
-    List<NodeTreeElement> findAllByNodeIdInIncludingTopicAndSubtopic(Set<Integer> nodeId);
+    @Query("""
+            SELECT DISTINCT nc
+            FROM NodeConnection nc
+            JOIN nc.parent p
+            JOIN nc.child c
+            LEFT JOIN nc.relevance rel
+            WHERE nc.parent.id IN :nodeId
+            AND ((:nodeTypes) IS NULL OR c.nodeType in :nodeTypes)
+            """)
+    List<NodeConnection> findAllByNodeIdInIncludingTopicAndSubtopic(Set<Integer> nodeId, List<NodeType> nodeTypes);
```

Comment:
Har du testa ytelsen på dette? Umiddelbart så tenker eg at det er en risiko for at hibernate må kjøre fleire spørringer for å hente nodeconnections enn før der det var henta et objekt med kun id og rank. Spesielt når du skriver JOIN og ikkje JOIN FETCH.

Reply:
Superenkel sammenligning: 
* Gammel kode (med alle ressurser) bruker ca 100ms på min pc. 
* Ny kode (som berre returnerer emner) bruker 150ms.
* Ny kode _med JOIN FETCH_ bruker 110ms.

Alle kall går mot curl http://localhost:8000/v1/subjects/urn:subject:1:5f50e974-6251-42c7-8f04-c4fabf395d0f/topics\?recursive\=true

Reply:
har ikke testa noe særlig, men da hørtes det ut som en no-brainer å bruke `JOIN FETCH` i alle fall 😄
Legger til 👍

Reply:
Testa å legge på joins på metadata, og fikk pressa tida litt til til litt under 100ms:
```
            SELECT DISTINCT nc
            FROM NodeConnection nc
            LEFT JOIN FETCH nc.metadata ncm LEFT JOIN FETCH ncm.grepCodes LEFT JOIN FETCH ncm.customFieldValues nccfv LEFT JOIN FETCH nccfv.customField
            JOIN FETCH nc.parent p
            LEFT JOIN FETCH p.metadata pm LEFT JOIN FETCH pm.grepCodes LEFT JOIN FETCH pm.customFieldValues pcfv LEFT JOIN FETCH pcfv.customField
            JOIN FETCH nc.child c
            LEFT JOIN FETCH c.metadata cm LEFT JOIN FETCH cm.grepCodes LEFT JOIN FETCH cm.customFieldValues ccfv LEFT JOIN FETCH ccfv.customField
            LEFT JOIN FETCH nc.relevance rel
            WHERE nc.parent.id IN :nodeId
            AND ((:nodeTypes) IS NULL OR c.nodeType in :nodeTypes)
            """)
```

Dette gir 18 spørringer mot basen, mens kun join på parent og child er 73 spørringer.

Reply:
Nice, da legger jeg til den! :smile: